src/HtmlUtils.tsx

```diff
@@ -601,6 +613,48 @@ export function bodyToHtml(content: IContent, highlights: string[], opts: IOpts
         </span>;
 }
 
+/* turn a room topic into html
+ *
+ * topic: plain text topic
+ * htmlTopic: optional html topic
+ * ref: React ref to attach to any React components returned
+ */
+export function topicToHtml(topic: string, htmlTopic?: string, ref?: React.Ref<HTMLSpanElement>): ReactNode {
+    if (!SettingsStore.getValue("feature_html_topic")) {
+        htmlTopic = null;
+    }
+
+    const isFormattedTopic = !!htmlTopic;
+    let topicHasEmoji = false;
+    let safeTopic = "";
+
+    try {
+        topicHasEmoji = mightContainEmoji(isFormattedTopic ? htmlTopic : topic);
+
+        if (isFormattedTopic) {
+            safeTopic = sanitizeHtml(htmlTopic, topicSanitizeHtmlParams);
+            if (topicHasEmoji) {
+                safeTopic = formatEmojis(safeTopic, true).join('');
+            }
+        }
+    } catch {}
+
+    let emojiBodyElements: JSX.Element[];
```

Comment:
```suggestion
    let emojiBodyElements: ReturnType<typeof formatEmojis>;
```

this should prevent the cast a couple lines later